src/server.rs

```diff
@@ -302,301 +258,168 @@ impl RollupBoostServer {
+            let payload = builder.get_payload_v3(payload_id).await?;
 
             // Send the payload to the local execution engine with engine_newPayload to validate the block from the builder.
             // Otherwise, we do not want to risk the network to a halt since op-node will not be able to propose the block.
             // If validation fails, return the local block since that one has already been validated.
-            let payload_status = self.l2_client.auth_client.new_payload_v3(payload.execution_payload.clone(), vec![], payload.parent_beacon_block_root).await.map_err(|e| {
-                error!(message = "error calling new_payload_v3 to validate builder payload", "url" = ?self.l2_client.auth_rpc, "error" = %e, "local_payload_id" = %payload_id, "external_payload_id" = %external_payload_id);
-                e
-            })?;
-            if let Some(mut s) = span {
-                s.end();
-            };
-            if let Some(mut parent) = parent_span {
-                let parent = Arc::get_mut(&mut parent);
-                if let Some(parent) = parent {
-                    parent.end();
-                }
-            };
+            let _ = self
+                .l2_client
+                .new_payload_v3(
+                    payload.execution_payload.clone(),
+                    vec![],
+                    payload.parent_beacon_block_root,
+                )
+                .await?;
 
-            if payload_status.is_invalid() {
-                error!(message = "builder payload was not valid", "url" = ?builder.auth_rpc, "payload_status" = %payload_status.status, "local_payload_id" = %payload_id, "external_payload_id" = %external_payload_id);
-                Err(ClientError::Call(ErrorObject::owned(
-                    INVALID_REQUEST_CODE,
-                    "Builder payload was not valid",
-                    None::<String>,
-                )))
-            } else {
-                info!(message = "received payload status from local execution engine validating builder payload", "local_payload_id" = %payload_id, "external_payload_id" = %external_payload_id);
-                Ok((payload, source))
-            }
+            Ok(Some(payload))
         });
 
         let (l2_payload, builder_payload) = tokio::join!(l2_client_future, builder_client_future);
-        let payload = match (builder_payload, l2_payload) {
-            (Ok(builder), _) => Ok(builder),
-            (Err(_), Ok(l2)) => Ok(l2),
-            (Err(_), Err(e)) => Err(e),
-        };
-        payload.map(|(payload, context)| {
-            let inner_payload = ExecutionPayload::from(payload.clone().execution_payload);
-            let block_hash = inner_payload.block_hash();
-            let block_number = inner_payload.block_number();
-
-            if let Some(metrics) = &self.metrics {
-                metrics.increment_blocks_created(&context);
```

Comment:
note that we also want to record whether the server used the builder or local payload here (e.g. if the builder payload is invalid, was discarded due to some other block selection mechanism) which looks to be removed from the instrumentation

Reply:
This value can be derived from the client get_payload_v3 spans. You can just look at spans where span kind is client, payload source is builder, and status is ok. A histogram with count is automatically created for this.

Reply:
the client get payload span only asserts that the response is ok, not necessarily what payload was chosen by rollup-boost. If the builder payload fails validation the metrics would not capture that the local payload was chosen

Reply:
This line:
```rust
        tracing::Span::current().record("payload_source", context.to_string());
```

Sets the payload source, which is added as a label.

Reply:
oh right, thanks for pointing that out! I think everything is at parity with the current implementation then